File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086ServerEventHandler.java
package org.emulinker.kaillera.controller.v086.action;

import org.emulinker.kaillera.controller.v086.V086Controller.V086ClientHandler;
import org.emulinker.kaillera.model.event.ServerEvent;

public interface V086ServerEventHandler<T extends ServerEvent> {
  @Override
  public String toString();

  public void handleEvent(T event, V086ClientHandler clientHandler);

  public int getHandledEventCount();
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086ServerEventHandler.kt
package org.emulinker.kaillera.controller.v086.action

import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.model.event.ServerEvent

interface V086ServerEventHandler<in T : ServerEvent> {
  @Deprecated("Structure this in a different way")
  override fun toString(): String

  suspend fun handleEvent(event: T, clientHandler: V086ClientHandler)
  val handledEventCount: Int
}
